packages/media/src/index.ts

```diff
@@ -1,12 +1,42 @@
 import { Accessor } from "solid-js";
 import { isServer } from "solid-js/web";
-import { makeEventListener } from "@solid-primitives/event-listener";
-import { entries, noop, createHydratableSignal } from "@solid-primitives/utils";
+import { entries, noop, createHydratableSignal, tryOnCleanup } from "@solid-primitives/utils";
 import { createHydratableStaticStore } from "@solid-primitives/static-store";
 import { createHydratableSingletonRoot } from "@solid-primitives/rootless";
 
 /**
- * attaches a MediaQuery listener to window, listeneing to changes to provided query
+ * Make listener helper to support binding events specifically for MQ.
+ * This function doesn't use the standard makeEventListener as iOS 13
+ * did not support addEventListener, hence this function checks for it
+ * before binding addListener instead.
+ * @param mq - ref to MediaQueryList
+ * @param type - name of the handled event
+ * @param handler - event handler
+ * @param options - addEventListener options
+ * @returns Function clearing all event listeners form targets
+ */
+type EventHandler<T> = (event: T) => void;
+export function makeListener(
+  mq: MediaQueryList,
+  type: string,
```

Comment:
Is there even a point in passing `type`? Wouldn't you just want `"change"`? If you pass something else, then you cannot call `addListener`.
Actually, in the body you are adding listener with `"change"` hardcoded and removing it with `type` param.